packages/modules/e3dc/inverter.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+from typing import Dict, Union
+import logging
+
+from dataclass_utils import dataclass_from_dict
+from modules.common import modbus
+from modules.common.component_state import InverterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.modbus import ModbusDataType, Endian
+from modules.common.fault_state import ComponentInfo
+from modules.common.store import get_inverter_value_store
+from modules.common.simcount._simcounter import SimCounter
+from modules.e3dc.config import E3dcInverterSetup
+
+
+log = logging.getLogger(__name__)
+
+
+def read_inverter(client: modbus.ModbusTcpClient_, read_ext):
+    # 40067 PV Leistung
+    pv = client.read_holding_registers(40067, ModbusDataType.INT_32,
+                                       wordorder=Endian.Little, unit=1) * -1
+    if read_ext == 1:
+        # 40075 externe PV Leistung
+        pv_external = client.read_holding_registers(40075, ModbusDataType.INT_32,
+                                                    wordorder=Endian.Little, unit=1)
```

Comment:
Da muss noch ein `else` dazu. Sonst geht es im Fall von `read_ext != 1` schief (es wird eine nicht existierende Variable gelesen).

Allerdings wird aktuell sowohl im Legacy-Fall, als auch im neuen Fall die beiden Werte nur zusammenaddiert. In dem Fall könntest du hier auch gleich nehmen:
```python
if read_ext == 1:
    pv += client.read_...
return pv
```
Es sei denn natürlich du splittest `read_ext` in eine separate Komponente auf.